src/base/torrentfilter.cpp

```diff
@@ -176,21 +177,26 @@ bool TorrentFilter::matchState(const BitTorrent::Torrent *const torrent) const
     case Inactive:
         return torrent->isInactive();
     case Stalled:
-        return (torrent->state() ==  BitTorrent::TorrentState::StalledUploading)
-                || (torrent->state() ==  BitTorrent::TorrentState::StalledDownloading);
+        return (state == BitTorrent::TorrentState::StalledUploading)
+                || (state == BitTorrent::TorrentState::StalledDownloading);
     case StalledUploading:
-        return torrent->state() ==  BitTorrent::TorrentState::StalledUploading;
+        return state == BitTorrent::TorrentState::StalledUploading;
     case StalledDownloading:
-        return torrent->state() ==  BitTorrent::TorrentState::StalledDownloading;
+        return state == BitTorrent::TorrentState::StalledDownloading;
     case Checking:
-        return (torrent->state() == BitTorrent::TorrentState::CheckingUploading)
-                || (torrent->state() == BitTorrent::TorrentState::CheckingDownloading)
-                || (torrent->state() == BitTorrent::TorrentState::CheckingResumeData);
+        return (state == BitTorrent::TorrentState::CheckingUploading)
+                || (state == BitTorrent::TorrentState::CheckingDownloading)
+                || (state == BitTorrent::TorrentState::CheckingResumeData);
     case Moving:
         return torrent->isMoving();
     case Errored:
         return torrent->isErrored();
+    case _Count:
```

Comment:
IMO, it's better to check for any unhandled values:
```suggestion
    default:
```

Reply:
It seems there are 2 ways for this issue (switching on enum types):
1. Let compiler emit warnings/errors for unhandled case
  https://github.com/qbittorrent/qBittorrent/actions/runs/8858078639/job/24326294606#step:9:401
  The coder will then add a new specific switch case for it.
2. use `default:` case and have failed assertions at run-time

I'm leaning towards the first way since it can be detected at compile-time rather than run-time.

Reply:
I mean that c++ enum variable can be assigned to "invalid" value (i.e. the "unnamed" one). I think it's good idea to add assertion on it.

Reply:
>I mean that c++ enum variable can be assigned to "invalid" value (i.e. the "unnamed" one). I think it's good idea to add assertion on it.

Done.
It seems that specific compiler warning was only useful in an imaginary world and will become useless in the real world. Such a pity...

Reply:
> It seems that specific compiler warning was only useful in an imaginary world and will become useless in the real world.

Some benefit from this warning could be gained if in the previous version of the code (the one without `default` label) add an assertion after `switch` block.

Reply:
>Some benefit from this warning could be gained if in the previous version of the code (the one without default label) add an assertion after switch block.

Right but there are cases that it cannot be applied so I choose the least intrusive change here.

Reply:
> I choose the least intrusive change here.

I don't object. But still... what do you mean by "there are cases that it cannot be applied"?

Reply:
>I don't object. But still... what do you mean by "there are cases that it cannot be applied"?

For example, (generally speaking) some switch case could be no-op and need to do other things following the switch statement. Then it would be wrong to have `assert(false)` after the switch statement.

Reply:
The problem with c++ enums seems to have no generic solution. But here's what I meant above:
1. If we add another filter type, but forget to process it in this `switch` block, then this could be detected at compile time (if we remove `default` label).
2. if we get an unsuitable value, for example, converted from some serialized form (from an integer, in the simplest case), then there are no options, this can only be detected at runtime.

The current code can detect both of these errors only at runtime, which increases the chance of geting the first type of error in release if it has not been tested in detail.

Reply:
> For example, (generally speaking) some switch case could be no-op and need to do other things following the switch statement. Then it would be wrong to have `assert(false)` after the switch statement.

Generally speaking, it doesn't seem like example of good code.
But let's not continue to evolve this topic. As I said, it does not have a generic solution, so there is no point in considering it from this angle.